lightning/src/chain/channelmonitor.rs

```diff
@@ -2008,33 +2008,37 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 			self.is_paying_spendable_output(&tx, height, &logger);
 		}
 
-		self.block_confirmed(height, txn_matched, watch_outputs, claimable_outpoints, broadcaster, fee_estimator, logger)
+		self.block_confirmed(height, txn_matched, watch_outputs, claimable_outpoints, &broadcaster, &fee_estimator, &logger)
 	}
 
+	/// Update state for new block(s)/transaction(s) confirmed. Note that the caller *must* update
+	/// `self.best_block` before calling, and set `conf_height` to the height at which any new
```

Comment:
Is this right? It isn't the case when called by `transactions_confirmed` as of now.

Reply:
I don't think it matters, but its probably simpler/nicer for users to actually set it there, so I did that.

Reply:
Is this behavior required to fix the bug? Adding it there is equivalent to calling `best_block_updated` after each `transactions_confirmed`. So doesn't this mean calling `best_block_updated` is no longer required when confirming transactions at the same block? Seems it is less simpler for users because it adds a conditional to `Confirm`'s contract, albeit optional.

Reply:
> Is this behavior required to fix the bug?

No.

> Adding it there is equivalent to calling best_block_updated after each transactions_confirmed. So doesn't this mean calling best_block_updated is no longer required when confirming transactions at the same block?

Yes, I believe so, at least in `ChannelMonitor`, dunno about `ChannelManager`.

> Seems it is less simpler for users because it adds a conditional to Confirm's contract, albeit optional.

I don't see why? It makes a call optional in our current implementation, yes, but we don't have to tell users that.

Reply:
> I don't see why? It makes a call optional in our current implementation, yes, but we don't have to tell users that.

It seems like surprising behavior. I called X expecting Y but instead Y + Z happened. I guess I don't see why not just drop the comment rather than altering the behavior if it isn't required to fix the bug. At very least it should be a separate commit stating the reason for the change, IMHO.

Reply:
It definitely feels "more correct" and should happen, IMO. I moved it to a new commit, let me know what you think of the commit docs and if it sufficiently justifies it.